src/vmspawn/vmspawn.c

```diff
@@ -818,6 +823,81 @@ static int kernel_cmdline_maybe_append_root(void) {
         return 0;
 }
 
+static int discover_bootloader(const char *root, char **ret_linux, char ***ret_initrds) {
+        _cleanup_(boot_config_free) BootConfig config = BOOT_CONFIG_NULL;
+        _cleanup_free_ char *esp_path = NULL, *xbootldr_path = NULL;
+        int r;
+
+        assert(root);
+        assert(ret_linux);
+        assert(ret_initrds);
+
+        esp_path = path_join(root, "efi");
+        if (!esp_path)
+                return log_oom();
+
+        xbootldr_path = path_join(root, "boot");
+        if (!xbootldr_path)
+                return log_oom();
```

Comment:
this is a bit sloppy, but dunno, maybe we can get away with this.

Reply:
I wasn't sure what an alternative would be to this as its a directory image we don't have PARTUUIDs to go off :(